### PR TITLE
Dfr 3175 CT controller postcode field not mandatory   amend application details

### PR DESCRIPTION
**Pull request by hamzahtahirhmcts**
### Jira link (if applicable)
[DFR-3175](https://tools.hmcts.net/jira/browse/DFR-3175)

### Checklist

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)
- [ ] Does this PR introduce a breaking change

### CODE REVIEW

File: src/test/resources/fixtures/contested/amend-applicant-solicitor-details-postcode-validation.json [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+      "detailPotentialAllegation" : "sfsffsd",
+      "benefitForChildrenDecision" : "No",
+      "paymentForChildrenDecision" : "Yes",
+      "addToComplexityListOfCourts" : "trueDontKnow",
+      "otherReasonForComplexityText" : "sdff",
+      "additionalPropertyOrderDecision" : "No",
+      "respondentRepresented" : "No"
+    }
+  }
+}
```
Comment (ptrelease) [inline src/test/resources/fixtures/contested/amend-applicant-solicitor-details-postcode-validation.json:140]: Add newline.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+            if (StringUtils.isBlank(applicantSolicitorPostCode)) {
+                errors.add("Postcode field is required for applicant solicitor address.");
+            }
+        }
+
+    }
+
+    private void validateApplicantPostcodeDetails(Map<String, Object> caseData, List<String> errors) {
+        String applicantPostCode = null;
```
Comment (ptrelease) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java:96]: You could remove this initializer if you wanted; copy what you did in validateApplicantSolicitorPostcodeDetails.

**Review by ptrelease — APPROVED** [review]

File: src/test/resources/fixtures/contested/amend-applicant-solicitor-details-postcode-validation.json (hunk 1 above)
Comment (hamzahtahirhmcts) [reply src/test/resources/fixtures/contested/amend-applicant-solicitor-details-postcode-validation.json]: [8984129](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/89841292ec84722782510a68fd6170ae30eaa877)

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        ((ObjectNode) requestContent.get("case_details").get("case_data").get("respondentAddress")).put("PostCode", "AB12 3CD");
+
+        mvc.perform(post(AMEND_APPLICATION_RES_SOL_URL)
+                .content(requestContent.toString())
+                .header(AUTHORIZATION_HEADER, AUTH_TOKEN)
+                .contentType(APPLICATION_JSON_VALUE))
+            .andExpect(status().isOk())
+            .andExpect(jsonPath("$.errors", hasSize(0)));
+    }
+}
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java:148]: add new line here.

Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java]: [0a0c58b](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/0a0c58bcce58e0406bde9ee6841c5a1e4eaf9b12)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+            String respondentPostCode = (String) respondentSolicitorAddress.get(postCode);
+
+            if (StringUtils.isBlank(respondentPostCode)) {
+                errors.add("Postcode field is required for respondent solicitor address.");
+            }
+        } else {
+            Map<String, Object> respondentAddress = (Map<String, Object>) caseData.get(RESPONDENT_ADDRESS);
+            String respondentPostCode = (String) respondentAddress.get(postCode);
+
+            if (StringUtils.isBlank(respondentPostCode)) {
```
[the remark this answers is not in the document]
Comment (hamzahtahirhmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java:132]: I'm pretty sure StringUtils.isBlank checks for null.

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java (hunk 2 above)
Comment (hamzahtahirhmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java]: [26f67a8](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/26f67a837783a0ebb5c7ff08b8b58c5b269e9228)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+public class AmendApplicationContestedController extends BaseController {
+
+    private static final String postCode = "PostCode";
+
+    @Autowired
+    private FinremCaseDetailsMapper finremCaseDetailsMapper;
+
+    private final InternationalPostalService postalService;
+
+
```
Comment (so99y) [inline src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java:48]: remove an extra blank line.

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.jsonPath;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.OrchestrationConstants.AUTHORIZATION_HEADER;
+
+@WebMvcTest(AmendApplicationContestedController.class)
+public class AmendApplicationContestedControllerTest extends BaseControllerTest {
+
+    private static final String AMEND_APPLICATION_APP_SOL_URL = "/case-orchestration/amend-application-validate-applicant-solicitor-address";
+    private static final String AMEND_APPLICATION_APP_URL = "/case-orchestration/amend-application-validate-applicant-address";
+    private static final String AMEND_APPLICATION_RES_SOL_URL = "/case-orchestration/amend-application-validate-respondent-solicitor";
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java:32]: Making use of uk.gov.hmcts.reform.finrem.caseorchestration.TestConstants.AUTH_TOKEN

File: src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+    public static final String AUTH_TOKEN = "tokien:)";
```
Comment (so99y) [inline src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java:33]: Making use of uk.gov.hmcts.reform.finrem.caseorchestration.TestConstants.AUTH_TOKEN

Comment (hamzahtahirhmcts) [reply src/test/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedControllerTest.java]: [510f05b](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/510f05b12c7681cbb84822cb3ed4f7051e199da6)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java (hunk 5 above)
Comment (hamzahtahirhmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java]: [510f05b](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/510f05b12c7681cbb84822cb3ed4f7051e199da6)

File: src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java [hunk 8]
```diff
@@ -0,0 +1,138 @@
...
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESP_SOLICITOR_ADDRESS;
+
+@RestController
+@RequestMapping(value = "/case-orchestration")
+@Slf4j
+@RequiredArgsConstructor
+public class AmendApplicationContestedController extends BaseController {
+
+    private static final String postCode = "PostCode";
```
[the remark this answers is not in the document]
Comment (hamzahtahirhmcts) [reply src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/controllers/AmendApplicationContestedController.java]: [510f05b](https://github.com/hmcts/finrem-case-orchestration-service/pull/1775/commits/510f05b12c7681cbb84822cb3ed4f7051e199da6)

**Review by so99y — APPROVED** [review]
LGTM

**Review by jthmcts — APPROVED** [review]

**Pull request merged**